pkg/core/hash.go

```diff
@@ -63,6 +63,96 @@ func calculateConfigHash(children []configObject) (string, error) {
 	return fmt.Sprintf("%x", hashBytes), nil
 }
 
+// updateHashBank stores hashes of each child in the in memory hashbank. It is
+// not threadsafe, and not to be called other than by a helper function.  as the
+// children should be uniquely named within the namespace it does not store the
+// instance name.
+func (h *Handler) updateHashBank(children []configObject) error {
+	// Add the hashes of each childs data to the hashBank
+	for _, child := range children {
+		if child.object != nil {
+			switch child.object.(type) {
+			case *corev1.ConfigMap:
+				hash, err := hashData(getConfigMapData(child))
+				if err != nil {
+					// log event and continue?
+					// return the error?
+				}
+				h.store.Bank[child.object.GetName()] = hashEntry{hash}
+			case *corev1.Secret:
+				hash, err := hashData(getSecretData(child))
+				if err != nil {
+					// log event and continue?
+				}
+				h.store.Bank[child.object.GetName()] = hashEntry{hash}
+			default:
+				return fmt.Errorf("passed unknown type: %v", reflect.TypeOf(child))
+			}
+		}
+	}
+	return nil
+}
+
+// initialiseHashBank checks if the instanceName exists in the hashBank, if not
+// it calls updateHashBank on the children.
+func (h *Handler) initialiseHashBank(instanceName string, children []configObject) error {
+	h.store.Lock()
+	defer h.store.Unlock()
+	if h.store.Bank == nil {
+		h.store.Bank = make(map[string]hashEntry)
+	}
+	if h.store.initialised == nil {
+		h.store.initialised = make(map[string]bool)
+	}
+
+	if _, ok := h.store.initialised[instanceName]; !ok {
```

Comment:
I'm also not sure if I need to keep this `initialised` map in memory. Could potentially just check if the existing instance hash is on the `podController`... although this would then fail if wave were restarted..